arch/x86/x86_features.c

```diff
@@ -100,6 +100,14 @@ void Z_INTERNAL x86_check_features(struct x86_cpu_features *features) {
         // check AVX512 bits if the OS supports saving ZMM registers
         if (features->has_os_save_zmm) {
             features->has_avx512 = ebx & 0x00010000;
+            if (features->has_avx512) {
```

Comment:
**suggestion:** 
- This comment can fit in 2 lines, we have line-length 120 in this repo.
- `has_avx512` should be renamed to `has_avx512f` to avoid being confusing.

Also, to simplify the checks throughout the code, perhaps it'd be good to add a features field for `has_avx512_common` or something like that, and have that depend on F, VL, DQ and BW.

Related to that, we could also (in a separate PR) add CD since that is AFAIK always included on cpus that implement F, and then allow the compiler to utilize the CD instructions if it wants to (collision detection for vectorizing loops). It might not actually be useful to us, but it could be interesting to look into whether the compiler chooses to use it for something, and it would be a logical thing to include in `has_avx512_common`.
Btw, we test making a compare256 variant that uses the avx512cd LZCNT instruction? It might be slower than AVX2 on early model cpus though. I still don't have access to any AVX512 cpus unfortunately.

Reply:
The lzcnt isn't as useful as it sounds, it will not count into the 512 bit vector, it will count per 32 or 64 bit lane.  Getting info out of that requires another reduction that has to check each individual value to know where the first nonzero bit occurred.  Additionally comparisons come back in mask registers, typically, at which point you really can't do _much_ fancy with it other than copy it back out to a GPR.  

Using march=native, it is using the masked comparisons and kortest, but I suspect it's not really going to buy us a ton.

Reply:
>    This comment can fit in 2 lines, we have line-length 120 in this repo.

Fixed.

>    has_avx512 should be renamed to has_avx512f to avoid being confusing.

Done

> Also, to simplify the checks throughout the code, perhaps it'd be good to add a features field for has_avx512_common or something like that, and have that depend on F, VL, DQ and BW.

Done.